### PR TITLE
add soft delete support completes #131

### PR DESCRIPTION
**Pull request by seancookr**
#131

### CODE REVIEW

File: lib/dolly/document.rb [hunk 1]
```diff
@@ -74,11 +74,15 @@ def save!
       save
     end
 
-    def destroy soft = false
-      #TODO: Add soft delete support
-      q = id_as_resource + "?rev=#{rev}"
-      response = database.delete(q)
-      JSON::parse response.parsed_response
+    def destroy hard = true
```
Comment (seancookr) [inline lib/dolly/document.rb:77]: personally I feel that `document.destroy(true)` should imply a full destroy, whereas `document.destroy(false)` should not.  Thus I changed the name of the arg

**Comment by javierg** [comment]
Is it possible to write a test?

**Comment by seancookr** [comment]
@javierg it is indeed possible

**Comment by javierg** [comment]
:shipit:

**Pull request merged**